src/js/Framework/FormCollection.js

```diff
@@ -51,6 +61,10 @@ const addItem = function (event, element) {
     removeItem(event)
   })
 
+  if (this.maximumItems && this.numberOfItems >= this.maximumItems) {
+    this.addButton.setAttribute('disabled', 'disabled')
+  }
+
   document.dispatchEvent(new Event('added.collection.item'))
```

Comment:
Maybe also check if this needs to be done. If they remove the disabled attribute from the button they still can add an item.

Reply:
Maybe even at the start of the method.